src/Lib.hs

```diff
@@ -46,22 +48,51 @@ sha256 = digestFromByteString . hash . pack
 -- abstracted hash function that takes a string
 -- to hash and returns a hex string
 hashString :: String -> String
-hashString = maybe (error "Something went wrong generating a hash") show . sha256
+hashString =
+  maybe (error "Something went wrong generating a hash") show . sha256
 
 calculateBlockHash :: Block -> String
-calculateBlockHash (Block i p t b _)  = hashString $ concat [show i, p, show t, b]
+calculateBlockHash (Block i p t b n _)  =
+  hashString $ concat [show i, p, show t, b, show n]
 
 -- returns a copy of the block with the hash set
 addHashToBlock :: Block -> Block
 addHashToBlock block = block { blockHash = calculateBlockHash block }
 
+-- Rudimentary proof-of-work: ensures that a block hash
+-- is less than a certain value (i.e. contains a certain
+-- amount of leading zeroes)
+-- In our case, it's 4 leading zeroes. We're using the Integer type
+-- since the current target is higher than the max for Int
```

Comment:
This is definitely good enough for people who are familiar with blockchains and PoW, but I feel like for those who are not, it could useful to briefly summarize what they are, why we have them, and how the nonce field ties in. Any other explanation you think would be helpful for that audience would be good!